### PR TITLE
8260633: [macos] java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.html test failed

### PR DESCRIPTION
**Pull request by alisenchung**
Opening closed dnd test
Test is green on all platforms

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8260633](https://bugs.openjdk.org/browse/JDK-8260633): [macos] java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.html test failed (**Bug** - P4)

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [8d63997e](https://git.openjdk.org/jdk/pull/19128/files/8d63997edbfd21e74a6b2c8f4f973366cb8801da)
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Committer) ⚠️ Review applies to [fb1d2e9b](https://git.openjdk.org/jdk/pull/19128/files/fb1d2e9b5ff7604ce39bdedbd3738741a323dd84)
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer) ⚠️ Review applies to [fb1d2e9b](https://git.openjdk.org/jdk/pull/19128/files/fb1d2e9b5ff7604ce39bdedbd3738741a323dd84)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19128/head:pull/19128` \
`$ git checkout pull/19128`

Update a local copy of the PR: \
`$ git checkout pull/19128` \
`$ git pull https://git.openjdk.org/jdk.git pull/19128/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19128`

View PR using the GUI difftool: \
`$ git pr show -t 19128`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19128.diff">https://git.openjdk.org/jdk/pull/19128.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19128#issuecomment-2099019451)

### CODE REVIEW

**Review by mrserb — APPROVED** [review]

File: test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        robot.waitForIdle();
+        robot.delay(MOUSE_RELEASE_TIMEOUT);
+
+        if (!passed) {
+            throw new RuntimeException("Test failed");
+        }
+    }
+
+    public void dispose() {
+        frame.dispose();
```
Comment (TejeshR13) [inline test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java:169]:
```suggestion
        SwingUtilities.invokeAndWait(() -> {
            if (frame != null) {
                frame.dispose();
            }
        }
```

File: test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            }
+        }
+    }
+
+    boolean pointInComponent(Robot robot, Point p, Component comp)
+            throws InterruptedException {
+        robot.waitForIdle();
+        reset();
+        robot.mouseMove(p.x, p.y);
+        robot.mousePress(InputEvent.BUTTON1_MASK);
```
Comment (TejeshR13) [inline test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java:190]:
```suggestion
        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
```

Comment (TejeshR13) [reply test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java:190]: Applies to other places too.

File: test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        return n < 0 ? -1 : n == 0 ? 0 : 1;
+    }
+
+    public void test() throws Exception {
+        final Robot robot = new Robot();
+        robot.setAutoDelay(45);
+        robot.waitForIdle();
+
+        final Point srcPoint = frame.getLocationOnScreen();
+        Dimension d = frame.getSize();
```
Comment (kumarabhi006) [inline test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java:132]: should be accessed on EDT.

File: test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @bug 4613903
+ * @summary verifies that mouse events are not dispatched during drag
+ * @key headful
+ * @run main MouseEventAfterStartDragTest
+ */
+public final class MouseEventAfterStartDragTest implements AWTEventListener {
```
Comment (DamonGuy) [inline test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java:53]:
```suggestion
 */
 
public final class MouseEventAfterStartDragTest implements AWTEventListener {
```

**Review by DamonGuy — APPROVED** [review]
LGTM. Just add a newline for separation after the test tags.

File: test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java (hunk 4 above)
Comment (alisenchung) [reply test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java:53]: i thought the standard was to have no newline between the test tags and the class declaration?

Comment (DamonGuy) [reply test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java:53]: I'm not aware of this. I've normally seen newlines after the test tags, and I use it in my tests without any comments. Not critical either way I suppose. Maybe others know more precisely? @TejeshR13 @kumarabhi006 maybe?

Comment (kumarabhi006) [reply test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java:53]: I use a newline after jtreg tags in my tests as well but I don't think it is as such mandatory.

**Review by TejeshR13 — APPROVED** [review]
LGTM.

**Comment by prrace** [comment]
You are opening the test, but the bug ID is about the test failing.
Why specifically did the test fail before and what about this version fixes that ?

File: test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java (hunk 4 above)
Comment (DamonGuy) [reply test/jdk/java/awt/dnd/MouseEventAfterStartDragTest/MouseEventAfterStartDragTest.java:53]: From the thread in the discussion Abhishek linked in our internal discussion, it looks like whatever we deem is neatest. But it looks like the big emphasis is on making it uniform. Sounds like most are using the newline.

**Comment by alisenchung** [comment]
> You are opening the test, but the bug ID is about the test failing. Why specifically did the test fail before and what about this version fixes that ?

The test needed some delays between mouse moves for stability. I accidentally deleted a delay when I moved the test but I've readded it now.

**Pull request closed without merging**